src/mechaphlowers/entities/arrays.py

```diff
@@ -0,0 +1,94 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pandas as pd
+import pandera as pa
+from pandera.typing import DataFrame, Series
+
+
+class ElementArray(ABC):
+    def __init__(self, data: DataFrame) -> None:
+        data = self._drop_extra_columns(data)
+        self.data = data
+
+    @property
+    @abstractmethod
+    def _input_columns(self) -> list[str]: ...
+
+    def _compute_extra_columns(self, input_data: DataFrame) -> list[str]:
+        return [
+            column for column in input_data.columns if column not in self._input_columns
+        ]
+
+    def _drop_extra_columns(self, input_data: DataFrame) -> DataFrame:
+        """Return a copy of the input DataFrame, without irrelevant columns.
+
+        Note: This has no impact on the input DataFrame.
+        """
+        extra_columns = self._compute_extra_columns(input_data)
+        return input_data.drop(columns=extra_columns)
+
+    def __str__(self) -> str:
+        return self.data.to_string()
+
+
+class SectionInputDataFrame(pa.DataFrameModel):
+    name: Series[str]
+    suspension: Series[bool]
+    conductor_attachment_altitude: Series[float]
+    crossarm_length: Series[float]
+    line_angle: Series[float]
+    insulator_length: Series[float]
+    span_length: Series[float] = pa.Field(nullable=True)
+
+    @pa.dataframe_check(
+        description="""Though tension supports also have insulators,
+        for now we ignore them when computing the state of a span or section.
+        Taking them into account might be implemented later.
+        For now, set the insulator length to 0 for tension supports to suppress this error."""
+    )
+    def insulator_length_is_zero_if_not_suspension(cls, df: DataFrame) -> Series[bool]:
```

Comment:
@adriengoeller For now the data validation raises an error if someone tries and instantiates a `SectionArray` from a dataframe which contains non-zero insulator lengths for one or more tension supports.

However I noticed it's possible to raise a [warning](https://pandera.readthedocs.io/en/stable/checks.html#raise-warning-instead-of-error-on-check-failure) instead of an error. I think this would be more appropriate - it would warn the user that some of the input data they provided won't be used in calculations, while not blocking them.

If you share this opinion, I suggest we open an issue about it so we can change the behavior when we have time.
What do you think?

Reply:
@rvduc what do you think ? We could also authorize positives values for tension support insulator but with no impact in calculation.

Reply:
At the very beginning, it isn't necessary to obtain an exact anchorage insulator length for sag-tension calculations. Therefore, to improve the precision of the calculation, it could be necessary to consider the actual insulator length and weight, as these factors may impact the true position of the cable attachment point. This means I believe it is better to be prepared to accept the anchorage insulator length.

Reply:
OK, thank you both for your feedback. 

Since we don't have that much time to work on it right now, I suggest we keep the current behavior for now and I'll open an issue to accept tension insulator lengths in input data and warn the user that we (still) don't take them into account in calculation.

We may take them into account later when we have time and when the needed equations (so to speak) are ready.

Reply:
Here's the issue: https://github.com/phlowers/mechaphlowers/issues/29

@adriengoeller I'll let you "prioritize" it when you think it's appropriate